#!/usr/bin/env python
PACKAGE = "controls"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("x", double_t, 0, "Tryphon : x", 3,  -7, 7)
gen.add("y", double_t, 0, "Tryphon : y", 0,  -7, 7)
gen.add("z", double_t, 0, "Tryphon : z",    2.0, 0, 7)
gen.add("roll", double_t, 0, "Tryphon : roll",  0 , -3.14,  3.14)
gen.add("pitch", double_t, 0, "Tryphon : pitch",  0 , -3.14,  3.14)
gen.add("yaw", double_t, 0, "Tryphon : yaw",  0 , -3.14,  3.14)
gen.add("fbuoy", double_t, 0, "Tryphon : ratio buoy/grav",  1 , 0.9995,  1.0005)
gen.add("path", bool_t, 0, "Tryphon : path",  False)
gen.add("pathNb", int_t, 0, "Tryphon : path number",  1, 1, 15)
gen.add("ctrlNb", int_t, 0, "Tryphon : ctrl number (1:LQR, 2:PID, 3:Computed torque)",  3, 1, 3)
gen.add("gaincp", double_t, 0, "Tryphon : gain computed torque",  0.3 , 0.05,  1)
gen.add("cd", double_t, 0, "Tryphon : drag coefficient",  0 , 0,  2.0)
gen.add("massTotal", double_t, 0, "Tryphon : mass total",  23 , 18,  28)
gen.add("maxThrust", int_t, 0, "Tryphon : max Thrust",  50, 10,  100)
gen.add("noInt", bool_t, 0, "Tryphon : no integral",  False)
gen.add("rend", int_t, 0, "Tryphon : path number", 0, 0, 2)
gen.add("onOff", bool_t, 0, "Tryphon : on",  True)


exit(gen.generate(PACKAGE, "controls", "control"))
